hail/python/hail/experimental/datasets.py

```diff
@@ -0,0 +1,294 @@
+                start: int32,
+                strand: int32,
+                transcript_consequences: array<struct {
+                    allele_num: int32,
+                    amino_acids: str,
+                    biotype: str,
+                    canonical: int32,
+                    ccds: str,
+                    cdna_start: int32,
+                    cdna_end: int32,
+                    cds_end: int32,
+                    cds_start: int32,
+                    codons: str,
+                    consequence_terms: array<str>,
+                    distance: int32,
+                    domains: array<struct {
+                        db: str,
+                        name: str
+                    }>,
+                    exon: str,
+                    gene_id: str,
+                    gene_pheno: int32,
+                    gene_symbol: str,
+                    gene_symbol_source: str,
+                    hgnc_id: str,
+                    hgvsc: str,
+                    hgvsp: str,
+                    hgvs_offset: int32,
+                    impact: str,
+                    intron: str,
+                    lof: str,
+                    lof_flags: str,
+                    lof_filter: str,
+                    lof_info: str,
+                    minimised: int32,
+                    polyphen_prediction: str,
+                    polyphen_score: float64,
+                    protein_end: int32,
+                    protein_start: int32,
+                    protein_id: str,
+                    sift_prediction: str,
+                    sift_score: float64,
+                    strand: int32,
+                    swissprot: str,
+                    transcript_id: str,
+                    trembl: str,
+                    uniparc: str,
+                    variant_allele: str
+                }>,
+                variant_class: str
+            }
+            'locus': locus<GRCh38>
+        ----------------------------------------
+        Entry fields:
+            'GT': call
+        ----------------------------------------
+        Column key: ['s']
+        Row key: ['locus', 'alleles']
+        Partition key: ['locus']
+        ----------------------------------------
+
+    Parameters
+    ----------
+    dataset_name : :obj:`str`
+        Name of the dataset to load.
+    reference_genome : `GRCh37` or `GRCh38`
+        Reference genome build.
+    config_file : :obj:`str`, optional
+        Path of the datasets configuration file.
+        Leave as default if running on Google Cloud Platform.
+
+    Returns
+    -------
+    :class:`.Table` or :class:`.MatrixTable`"""
+
+    with hl.hadoop_open(config_file, 'r') as f:
+        config = json.load(f)
+
+    builds = [{'path': x['path'], 'reference_genome': x['reference_genome']}
+              for x in config if x['name'] == dataset_name]
+
+    if not builds:
+        raise NameError("Dataset '{}' not found.".format(dataset_name))
```

Comment:
NameError refers to a variable not being in `locals` or `globals`, this should be a `KeyError`